internal/controller/pkg/manager/reconciler.go

```diff
@@ -329,10 +330,35 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		return reconcile.Result{}, err
 	}
 
-	imageConfig, pullSecretFromConfig, err := r.config.PullSecretFor(ctx, p.GetSource())
+	// Rewrite the image path if necessary. We need to do this before looking
+	// for pull secrets, since the rewritten path may use different secrets than
+	// the original.
+	imagePath := p.GetSource()
+	rewriteConfigName, newPath, err := r.config.RewritePath(ctx, imagePath)
+	if err != nil {
+		err = errors.Wrap(err, errRewriteImage)
+		p.SetConditions(v1.Unpacking().WithMessage(err.Error()))
+		p.ClearAppliedImageConfigRef(v1.ImageConfigReasonRewrite)
```

Comment:
Is this the right place to clear the ref? The controller's just resolving the path here, right? It hasn't persisted the applied config ref yet (at least not on this reconcile).